assessments/projects/openkruise/self-assessment.md

```diff
@@ -0,0 +1,656 @@
+
+### Non-goals
+
+#### General
+
+- OpenKruise is **not** a PaaS and it will **not** provide any abilities of PaaS
+- Providing ways of managing containers without Kubernetes
+- Replacement for kubernetes: OpenKruise is not intended as a means to replace
+  kubernetes, rather it is an extension to Kubernetes providing extra features
+  on top.
+- Standardization of deployment strategies: OpenKruise introduces several
+  advanced deployment strategies, it does not aim to standardize these
+  strategies across Kubernetes cluster. Users are free to manipulate the
+  strategy based on their needs.
+
+#### Security
+
+- OpenKruise doesn't provide additional security for Kubernetes
+- No guarantee of security when an attacker has compromised the system
+
+## Self-assessment use
+
+This self-assessment is created by the OpenKruise team to perform an internal
+analysis of the project's security. It is not intended to provide a security
+audit of OpenKruise, or function as an independent assessment or attestation
+of OpenKruise's security health.
+
+This document serves to provide OpenKruise users with an initial understanding
+of OpenKruise's security, where to find existing security documentation,
+OpenKruise plans for security, and general overview of OpenKruise security
+practices, both for development of OpenKruise as well as security of
+OpenKruise.
+
+This document provides the CNCF TAG-Security with an initial understanding of
+OpenKruise to assist in a joint-assessment, necessary for projects under
+incubation. Taken together, this document and the joint-assessment serve as a
+cornerstone for if and when OpenKruise seeks graduation and is preparing for a
+security audit.
+
+## Security functions and features
+
+### Critical
+
+- Security scanning with Snyk in the CI pipeline identifies vulnerabilities in
+  container images so only verified images are displayed.
+- Security scanning with CodeQL in the CI pipeline identifies variants of known
+  security vulnerability in the codebase.
+- Supporting only recent software versions that provide patches and updates
+  mitigates general vulnerabilities.
+
+### Security Relevant
+
+- Regularly scanning the code in the main (master) and nightly builds, as well
+  as in pull requests (PRs) for the Go programming language helps identify any
+  potential vulnerabilities or issues before release.
+- Scanning the container images that are published on the GitHub Container
+  Registry ensures that the images, which are used to run OpenKruise in a
+  Kubernetes environment, are secure.
+
+### Threat Modeling with STRIDE
+
+#### Spoofing:
+
+- Threat-01-S: Impersonating an Administrator
+- An attacker could impersonate an OpenKruise admin, maintainer, or somebody
+  with elevated privilege in order to access data within the containers, code
+  repositories, or contributions from other developers.
+
+##### Recommended Mitigations
+
+- Assign proper roles (per principle of least privilege) to the users using
```

Comment:
Thanks for raising this point. Let me take a closer look at get back to you here.

Reply:
So to clarify bit here, the threat model is OpenKruise specific. When it comes to some of the more general points in the threat model like the example you listed around RBAC in Kubernetes, we would assume that they are following the principle of least privilege here but we we're able to prove that it was in fact the case. For that reason we've made sure to explicitly list it as a point of concern here since it's not clear for the project. The same would go for a few of the other more general points.

We have had one of the maintainers take a look at the entire threat model so my assumption is also that if this wasn't flagged as irrelevant then it makes sense to keep. What are your thoughts on how to proceed there?

Reply:
Got it. Glad to hear maintainers have reviewed this work and are good with it. Sounds good to me.
Following discussions among the review team, we are asking that all STRIDE content be moved to a secondary document within this directory. Please extract this content and add a link to the new file in the appendix.